### DISCUSSION TITLE
hide videos based on title using regular expressions

### DISCUSSION THREAD

**Discussion by ticmaisdev** [FreeTubeApp/FreeTube#4572] [General]
Currently, we are able to hide videos from channels based on:

1. **exact** channel name
2. channel ID

But I judge this to not be as broad as it should be.

My suggestion is making it possible so that users can use regular expressions to hide videos based on:

- regex for title

My personal use cases would be for:

- [1 hour long version of music videos](https://youtu.be/ckgTbs-Cl14)

It's very frustrating to be disrupted from work because autoplay ends up on one of these videos playing the same song in a loop. Not to talk about those channels making huge playlists of songs that I only really enjoy listening to a couple them.

- https://www.youtube.com/live/v70VhIeurKQ

I don't think this should be difficult to implement, considering that we already have:

1. UI for similar features
2. logic for similar features

![image](https://github.com/FreeTubeApp/FreeTube/assets/128381407/a93684a1-81fa-4c30-b545-536f8fb9f8dc)

**Comment by ticmaisdev** [comment]
Please, remember to click on "vote" button right after chosing the only possible option.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
This will be addressed by https://github.com/FreeTubeApp/FreeTube/pull/4202
